// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// If you change or add any fields in this file, update proto_visitors.h and
// potentially proto_enum_conversions.{h, cc}.

syntax = "proto2";

option java_multiple_files = true;
option java_package = "org.chromium.components.sync.protocol";

option optimize_for = LITE_RUNTIME;

package sync_pb;

// The payload coming from the server for all invalidated data types.
message SyncInvalidationsPayload {
  message DataTypeInvalidation {
    // The legitimate values of this field correspond to the protobuf field
    // numbers of all EntitySpecifics fields supported by the server (see
    // components/sync/protocol/sync.proto).
    optional int32 data_type_id = 1;
  }

  // This field contains information about each data type which needs to be
  // updated. Note that it may contain data types which current client is not
  // interested in.
  repeated DataTypeInvalidation data_type_invalidations = 1;

  // Opaque field, which has to be provided as part of resulting GetUpdates
  // back to the server.
  optional bytes hint = 2;

  // Version of invalidation, used to order incoming invalidations.
  optional int64 version = 3;
}
